### PR TITLE
[Diagnostics] Adding Propagate Exceptions Snippets to ASP.NET Core

### PR DESCRIPTION
**Pull request by amanda-tarafa**
For helping troubleshoot issues similar to #2630.
(Diagnostics ASP.NET does not offer explicit Logging, only Error Reporting and Trace, so this does not apply)

### CODE REVIEW

File: apis/Google.Cloud.Diagnostics.AspNetCore/docs/index.md [hunk 1]
```diff
@@ -64,6 +68,29 @@ property on the `Startup` class to obtain values for `ProjectID`, `Service` or `
...
 
+## Troubleshooting Logging
+
+Sometimes it is neccessary to diagnose log operations. It might be that logging is failing or that
+we simply cannot find where the logs are being stored in GCP. What follows are a couple of code samples
+that can be useful to find out what might be wrong with logging operations.
+
+### Propagating Exceptions
+
+By default Google Logger won't propagate any exceptions thrown during logging. This is to protect the
```
Comment (jskeet) [inline apis/Google.Cloud.Diagnostics.AspNetCore/docs/index.md]: Add "the" after "by default" perhaps?

File: apis/Google.Cloud.Diagnostics.AspNetCore/docs/index.md [hunk 2]
```diff
@@ -64,6 +68,29 @@ property on the `Startup` class to obtain values for `ProjectID`, `Service` or `
 
 {{sample:Logging.UseGoogleLogger}}
 
+## Troubleshooting Logging
+
+Sometimes it is neccessary to diagnose log operations. It might be that logging is failing or that
+we simply cannot find where the logs are being stored in GCP. What follows are a couple of code samples
+that can be useful to find out what might be wrong with logging operations.
+
+### Propagating Exceptions
```
Comment (jskeet) [inline apis/Google.Cloud.Diagnostics.AspNetCore/docs/index.md]: I'm not sure whether having a whole extra section here is worth it - maybe just promote the two samples to `###` level, and include the paragraph below directly in this section? (i.e. just remove this header and one `#` from each of the headers below).

File: apis/Google.Cloud.Diagnostics.AspNetCore/docs/index.md (hunk 1 above)
Comment (amanda-tarafa) [reply apis/Google.Cloud.Diagnostics.AspNetCore/docs/index.md]: Done.

File: apis/Google.Cloud.Diagnostics.AspNetCore/docs/index.md (hunk 2 above)
Comment (jskeet) [reply apis/Google.Cloud.Diagnostics.AspNetCore/docs/index.md]: Sure, no problem.

**Comment by amanda-tarafa** [comment]
Thanks, waiting on green and merging.

**Pull request merged**